code/modules/hippie/items/devices/traitordevices.dm

```diff
@@ -0,0 +1,82 @@
+/* Research Director Syndie teleporter */
+/obj/item/device/experimental_teleporter
+	name = "experimental teleporter"
+	desc = "An imperfect teleportation device."
+	icon = 'icons/hippie/obj/device.dmi'
+	icon_state = "spe_teleporter"
+	force = 0
+	throwforce = 0
+	w_class = 2
+	throw_speed = 3
+	throw_range = 5
+	materials = list(MAT_METAL=10000)
+	origin_tech = "magnets=4;bluespace=6;syndicate=5"
+	var/cooldown = 6 /* 0.6 sec cd */
+	var/active = TRUE
+
+/obj/item/device/experimental_teleporter/attack_self(mob/user)
+	if(!active)
+		return
+	active = FALSE
+
+	var/datum/effect_system/spark_spread/spark_system = new /datum/effect_system/spark_spread
+	spark_system.set_up(5, 0, src)
+	spark_system.start()
+
+	playsound(get_turf(user), 'sound/hippie/effects/teleport.ogg', 40, 1)
+
+	user.visible_message("<span class='warning'>[user.name] teleports away!</span>")
+
+	/* Teleport. If they go out of bounds, then stop processing */
+	var/turf/newTurf = teleport(user)
+	if(!newTurf)
+		return
+
+	if(isclosedturf(newTurf)) /* If you get stuck in a closed turf, you die */
+		user.emote("scream")
+		user << "<span class='userdanger'>You scream out in pain as your body materialises inside of the [newTurf.name]!</span>"
+		active = TRUE /* so that it's not perma broke */
+		user.gib()
+		return
+
+	for(var/mob/living/victim in newTurf)
+		if(victim == user)
+			continue
+		if(iscarbon(victim)) /* Robots shouldn't scream in pain */
+			victim.emote("scream")
+			victim << "<span class='userdanger'>You scream out in pain as your body is ripped apart by [user.name]!</span>"
+		else
+			victim << "<span class='userdanger'>Your chassis is ripped apart by [user.name]!</span>"
+		victim.gib()
+
+	addtimer(CALLBACK(src, .proc/reactivate), cooldown)
```

Comment:
Any examples on how to do that? Do I just keep a reference to the timer and remove the timer on object qdel?